packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -3138,35 +3148,32 @@ export default class Meeting extends StatelessWebexPlugin {
 
   /**
    * Sets the local media stream on the class and emits an event to the developer
-   * @param {MediaStream} localShare the local media stream
+   * @param {MediaStreamTrack} localShareTrack the local media stream
    * @returns {undefined}
    * @public
    * @memberof Meeting
    */
-  public setLocalShareTrack(localShare: MediaStream) {
+  public setLocalShareTrack(localShareTrack: MediaStreamTrack) {
     let settings = null;
 
-    if (localShare) {
-      this.mediaProperties.setLocalShareTrack(MeetingUtil.getTrack(localShare).videoTrack);
-      const contentTracks = this.mediaProperties.shareTrack;
-
-      if (contentTracks) {
-        settings = contentTracks.getSettings();
-        this.mediaProperties.setMediaSettings('screen', {
-          aspectRatio: settings.aspectRatio,
-          frameRate: settings.frameRate,
-          height: settings.height,
-          width: settings.width,
-          displaySurface: settings.displaySurface,
-          cursor: settings.cursor,
-        });
-        LoggerProxy.logger.log(
-          'Meeting:index#setLocalShareTrack --> Screen settings.',
-          JSON.stringify(this.mediaProperties.mediaSettings.screen)
-        );
-      }
+    if (localShareTrack) {
+      this.mediaProperties.setLocalShareTrack(localShareTrack);
+
+      settings = localShareTrack.getSettings();
+      this.mediaProperties.setMediaSettings('screen', {
+        aspectRatio: settings.aspectRatio,
+        frameRate: settings.frameRate,
+        height: settings.height,
+        width: settings.width,
+        displaySurface: settings.displaySurface,
+        cursor: settings.cursor,
+      });
+      LoggerProxy.logger.log(
+        'Meeting:index#setLocalShareTrack --> Screen settings.',
+        JSON.stringify(this.mediaProperties.mediaSettings.screen)
+      );
 
-      contentTracks.onended = () => this.handleShareTrackEnded(localShare);
+      localShareTrack.addEventListener('ended', this.handleShareTrackEnded);
```

Comment:
this will change when on ended event gets added to track

Reply:
yes. This code will be updated to use the "LocalTrack ended" event from webrtc-core when we use the new webrtc-core classes